src/net/telnet.rs

```diff
@@ -30,11 +32,33 @@ impl TelnetHandler {
             main_writer: session.main_writer,
             output_buffer: session.output_buffer,
             mode: TelnetMode::UnterminatedPrompt,
+            will_ga: false,
+            will_eor: false,
         }
     }
-}
 
-impl TelnetHandler {
+    fn determine_telnet_mode(&mut self) {
```

Comment:
small naming nit: `determine` makes me think it would be idempotent and return the new telnet mode as opposed to mutating `self` and setting the mode for `self` and `output_buffer`.

```suggestion
    fn update_telnet_mode(&mut self) {
```

Reply:
Sure